src/server/formatting.rs

```diff
@@ -24,7 +28,7 @@ pub fn get_formatting_unregistration() -> Unregistration {
 impl TypstServer {
     pub fn format_document(&self, source: &Source) -> anyhow::Result<Vec<TextEdit>> {
         let original_text = source.text();
-        let res = typstfmt_lib::format(original_text, typstfmt_lib::Config::default());
+        let res = typstfmt_lib::format(original_text, self.get_fmt_config());
```

Comment:
Since `get_fmt_config` reads from the filesystem, it can take a while to run. I'd prefer to cache the config, ideally via the abstractions in `workspace/fs`.

Reply:
Hmm, I figured since we're just reading 1 file, that might make sense, since otherwise we need to watch the file.... I'll have a look at `workspace/fs` to see if I can figure that out, thanks!

Reply:
I've looked at `workspace/fs/cache`, but it does not seem to provide invalidating a file entry when the file changes on disk externally. I'd consider this undesirable, so instead I made the IO async.

Reply:
I'm not sure that async IO solves the problem that I think exists. I'll need a little more time to review this code, so I'm going to put off merging this until I can sit down and take a closer look.

Reply:
Note that I pushed an update, because I think errors from reading toml should be surfaced. Not sure why neovim doesn't show them, but they show up in the lsp log, so fine for now :)

If there's anything I can help with let me know (I'm not sure what the problem is that might exist), but otherwise, take your time of course :)